src/BlackSlope.Api/Startup.cs

```diff
@@ -1,104 +1,100 @@
-            HostConfig = serviceProvider.GetService<HostConfig>();
-        }
-
-        public static IMapper GenerateMapperConfiguration()
-        {
-            var config = new MapperConfiguration(cfg =>
-            {
-                cfg.AddProfiles("BlackSlope.Hosts.Api");
-                cfg.AddProfiles("BlackSlope.Services.MovieService");
-                cfg.AddProfiles("BlackSlope.Repositories.MovieRepository");
-            });
-            return config.CreateMapper();
-        }
-
-        private void CorsConfiguration(IServiceCollection services)
-        {
-            services.AddCors(options =>
-            {
-                options.AddPolicy("AllowSpecificOrigin",
-                    builder => builder.AllowAnyOrigin()     // TODO: Replace with FE Service Host as appropriate to constrain clients
-                        .AllowAnyHeader()
-                        .WithMethods("PUT", "POST", "OPTIONS", "GET", "DELETE"));
-            });
-        }
-
-        private void AuthenticationConfiguration(IServiceCollection services)
-        {
-            services.AddAuthentication(options =>
-            {
-                options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
-            })
-            .AddJwtBearer(options =>
-            {
-                options.Authority = string.Format(HostConfig.AzureAd.AadInstance, HostConfig.AzureAd.Tenant);
-                options.Audience = HostConfig.AzureAd.Audience;
-            });
-        }
-    }
-}
+﻿using System.IO.Abstractions;
+using System.Reflection;
+using BlackSlope.Api.Common.Configurtion;
+using BlackSlope.Api.Common.Extensions;
+using BlackSlope.Api.Common.Middleware.Correlation;
+using BlackSlope.Api.Common.Middleware.ExceptionHandling;
+using BlackSlope.Api.Common.Version.Interfaces;
+using BlackSlope.Api.Common.Version.Services;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace BlackSlope.Api
+{
+    public class Startup
+    {
+        private readonly IConfiguration _configuration;
+        private HostConfig HostConfig { get; set; }
+
+        public Startup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvcService();
+            ApplicationConfiguration(services);
+            CorsConfiguration(services);
+
+            services.AddSwagger(HostConfig.Swagger);
+            services.AddAzureAd(HostConfig.AzureAd);
+            services.AddAutoMapper();
+            services.AddCorrelation();
+            services.AddTransient<IFileSystem, FileSystem>();
+            services.AddTransient<IVersionService, AssemblyVersionService>();
+            services.AddMovieService();

```

Comment:
I kind of liked the new lines we had previously to separate the cross cutting with the domain related items here

Reply:
Added the new line back.